<?xml version="1.0" ?>
<!-- TM5-900 URDF Example -->
<robot name="tm5-900">

  <link name="link_0">
    <visual>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/visual/tm5-base.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/collision/tm5-base_c.stl"/>
      </geometry>
    </collision>
  </link>

  <joint name="shoulder_1_joint" type="revolute">
    <parent link="link_0"/>
    <child link="link_1"/>
    <origin rpy="0.000000 -0.000000 0.000000" xyz="0.000000 0.000000 0.145200"/>
    <axis xyz="0 0 1"/>
    <limit effort="157" lower="-4.71238898038469" upper="4.71238898038469" velocity="3.141592653589793"/>
  </joint>

  <link name="link_1">
    <visual>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/visual/tmr_400w_01.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/collision/tmr_400w_01_c.stl"/>
      </geometry>
    </collision>
  </link>

  <joint name="shoulder_2_joint" type="revolute">
    <parent link="link_1"/>
    <child link="link_2"/>
    <origin rpy="-1.570796 -1.570796 0.000000" xyz="0.000000 0.000000 0.000000"/>
    <axis xyz="0 0 1"/>
    <limit effort="157" lower="-3.141592653589793" upper="3.141592653589793" velocity="3.141592653589793"/>
  </joint>

  <link name="link_2">
    <visual>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/visual/tm5-900_arm1.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/collision/tm5-900_arm1_c.stl"/>
      </geometry>
    </collision>
  </link>

  <joint name="elbow_joint" type="revolute">
    <parent link="link_2"/>
    <child link="link_3"/>
    <origin rpy="0.000000 -0.000000 0.000000" xyz="0.429000 0.000000 0.000000"/>
    <axis xyz="0 0 1"/>
    <limit effort="157" lower="-2.705260340591211" upper="2.705260340591211" velocity="3.141592653589793"/>
  </joint>

  <link name="link_3">
    <visual>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/visual/tm5-900_arm2.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/collision/tm5-900_arm2_c.stl"/>
      </geometry>
    </collision>
  </link>

  <joint name="wrist_1_joint" type="revolute">
    <parent link="link_3"/>
    <child link="link_4"/>
    <origin rpy="0.000000 -0.000000 1.570796" xyz="0.411500 0.000000 -0.122300"/>
    <axis xyz="0 0 1"/>
    <limit effort="43" lower="-3.141592653589793" upper="3.141592653589793" velocity="3.9269908169872414"/>
  </joint>

  <link name="link_4">
    <visual>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/visual/tmr_100w_01.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/collision/tmr_100w_01_c.stl"/>
      </geometry>
    </collision>
  </link>

  <joint name="wrist_2_joint" type="revolute">
    <parent link="link_4"/>
    <child link="link_5"/>
    <origin rpy="1.570796 -0.000000 0.000000" xyz="0.000000 -0.106000 0.000000"/>
    <axis xyz="0 0 1"/>
    <limit effort="43" lower="-3.141592653589793" upper="3.141592653589793" velocity="3.9269908169872414"/>
  </joint>

  <link name="link_5">
    <visual>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/visual/tmr_100w_02.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/collision/tmr_100w_02_c.stl"/>
      </geometry>
    </collision>
  </link>

  <joint name="wrist_3_joint" type="revolute">
    <parent link="link_5"/>
    <child link="link_6"/>
    <origin rpy="1.570796 -0.000000 0.000000" xyz="0.000000 -0.113150 0.000000"/>
    <axis xyz="0 0 1"/>
    <limit effort="43" lower="-4.71238898038469" upper="4.71238898038469" velocity="3.9269908169872414"/>
  </joint>

  <link name="link_6">
    <visual>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/visual/tmr_ee.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://tm_description/meshes/tm5-900/collision/tmr_ee_c.stl"/>
      </geometry>
    </collision>
  </link>

  <joint name="flange_fixed_joint" type="fixed">
    <parent link="link_6"/>
    <child link="flange_link"/>
    <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
  </joint>

  <link name="flange_link"/>

  <link name="base"/>
  <joint name="base_fixed_joint" type="fixed">
    <parent link="base"/>
    <child link="link_0"/>
    <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
  </joint>

  <link name="tool0"/>
  <joint name="flange_link-tool0" type="fixed">
    <parent link="flange_link"/>
    <child link="tool0"/>
    <origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
  </joint>

</robot>
